packages/shorebird_cli/lib/src/platform/apple.dart

```diff
@@ -110,6 +110,60 @@ Apple get apple => read(appleRef);
 
 /// A class that provides information about the iOS platform.
 class Apple {
+  /// Copies the supplement files into the build directory.
+  /// Currently we run gen_snapshot from `flutter`, both for the release and
+  /// patch builds. Both times it produces supplement files in a directory.
+  /// In the release case, these files are zipped up and stored as an artifact
+  /// on our servers for later use. In the patch case, they were created on
+  /// disk just before this call.
+  /// In both cases we need to copy the supplement files from these directories
+  /// to right next to where the snapshot files are before calling into
+  /// `aot_tools` to link the two snapshots together.
+  // TODO(eseidel): We should pass the entire supplement directories to
+  // `aot_tools` rather than having to know the contents within `shorebird`.
+  void copySupplementFilesToSnapshotDirs({
+    required Directory releaseSupplementDir,
+    required Directory releaseSnapshotDir,
+    required Directory patchSupplementDir,
+    required Directory patchSnapshotDir,
+    String patchSnapshotBaseName = 'out',
+  }) {
+    // All known supplement files names seen across all Flutter versions.
+    final supplementFileNames = <String>[
+      'App.ct.link',
+      'App.class_table.json',
+      'App.ft.link',
+      'App.field_table.json',
+    ];
+
+    // This uses maybeCopy because not all versions of gen_snapshot/aot_tools
+    // use the same supplement files. At the `shorebird` level we don't know
+    // which files should be present, so we just try to copy all.
+    void maybeCopy(File file, Directory destDir, {String? newBaseName}) {
+      if (!file.existsSync()) return;
```

Comment:
Aren't we checking if file exists twice (once here and once before calling `maybeCopy`)? We should probably remove one of the two (it's probably simpler to remove the one from the caller).

Reply:
Will do.